packages/core/src/util/runtime.ts

```diff
@@ -177,3 +180,41 @@ export const isEnabled = (
 
   return true;
 };
+
+export const isInherentlyEnabled = (
+  state: JsonFormsState,
+  ownProps: any,
+  uischema: UISchemaElement,
+  schema: JsonSchema & { readonly?: boolean; readOnly?: boolean },
+  rootData: any,
+  config: any
+) => {
+  if (state?.jsonforms?.readonly) {
+    return false;
+  }
+  if (uischema && hasEnableRule(uischema)) {
+    return isEnabled(uischema, rootData, ownProps?.path, getAjv(state));
+  }
+  if (typeof uischema?.options?.readonly === 'boolean') {
+    return !uischema.options.readonly;
+  }
+  if (typeof uischema?.options?.readOnly === 'boolean') {
+    return !uischema.options.readOnly;
+  }
+  if (typeof config?.readonly === 'boolean') {
+    return !config.readonly;
+  }
+  if (typeof config?.readOnly === 'boolean') {
+    return !config.readOnly;
+  }
+  if (typeof schema?.readOnly === 'boolean') {
+    return !schema.readOnly;
+  }
+  if (typeof schema?.readonly === 'boolean') {
+    return !schema.readonly;
+  }
```

Comment:
The schema specification only defines `readOnly` so we should not react on `readonly`
```suggestion
```